pika/connection.py

```diff
@@ -1102,14 +1149,15 @@ def _deliver_frame_to_channel(self, value):
 
         """
         if not value.channel_number in self._channels:
-            if self._is_basic_deliver_frame(value):
-                self._reject_out_of_band_delivery(value.channel_number,
-                                                  value.method.delivery_tag)
-            else:
-                LOGGER.warning("Received %r for non-existing channel %i", value,
-                               value.channel_number)
+            # This should never happen and would constitute breach of the
+            # protocol
+            LOGGER.critical(
+                'Received %s frame for unregistered channel %i on %s',
+                value.NAME, value.channel_number, self)
```

Comment:
REVIEWME There should be no such thing as "out-of-band delivery". It it happens, something is seriously wrong with either the broker's or pika's implementation.